tests/utils/profilehandler/profile_handler.go

```diff
@@ -685,3 +685,172 @@ func CreateClusterByProfile(profile *Profile, client *rosacli.Client, waitForClu
 	}
 	return description, err
 }
+
+func WaitForClusterUninstalled(client *rosacli.Client, cluster string, timeoutMin int) error {
+
+	endTime := time.Now().Add(time.Duration(timeoutMin) * time.Minute)
+	for time.Now().Before(endTime) {
+		output, err := client.Cluster.DescribeCluster(cluster)
+		desc, err := client.Cluster.ReflectClusterDescription(output)
+
+		if err != nil {
+			return err
+		}
+		if strings.Contains(output.String(), fmt.Sprintf("There is no cluster with identifier or name '%s'", cluster)) {
+			log.Logger.Infof("Cluster %s has been deleted.", cluster)
+			return nil
+		}
+		if strings.Contains(desc.State, con.Uninstalling) {
+			time.Sleep(2 * time.Minute)
+			continue
+		}
+		return fmt.Errorf("Cluster %s is in status of %s which won't be deleted, stop waiting", cluster, desc.State)
+	}
+	return fmt.Errorf("timeout for waiting for cluster deletion finished after %d mins", timeoutMin)
+}
+
+func DestroyClusterByProfile(profile *Profile, client *rosacli.Client, waitForClusterUninstall bool) (errors []error) {
```

Comment:
sometimes we want to just cleanup user data ,like resource created by function like "GenerateCommandByProfile" which will not create cluster. It's better to rename this function like 'CleanResourcesByProfile'.
Or separating the parts to clean user data into another function is better. Then DestroyClusterByProfile can use the separated function, if also more clear about the flow which clean some resources must be after the cluster is uninstalled.